File: DevExpress-Demo/VB/GridDemo.Wpf/Controls/SalesByYearData.vb
Imports DevExpress.DemoData.Models
Imports System
Imports System.Collections.Generic
Imports System.Dynamic

Namespace GridDemo

    Public Module SalesByYearData

        Private _Columns As List(Of GridDemo.ColumnDescription)

        Public ReadOnly Property Data As List(Of ExpandoObject)
            Get
                Dim lData = New List(Of ExpandoObject)()
                Dim random As Random = New Random()
                For yearIndex As Integer = 10 To 0 + 1 Step -1
                    Dim year As Integer = Date.Now.Year - yearIndex
                    For month As Integer = 1 To 12
                        Dim row As IDictionary(Of String, Object) = New ExpandoObject()
                        row("Date") = New DateTime(year, month, 1)
                        For columnIndex As Integer = 0 To Columns.Count - 1
                            row(Columns(columnIndex).PropertyName) = random.Next(30000)
                        Next

                        lData.Add(CType(row, ExpandoObject))
                    Next
                Next

                Return lData
            End Get
        End Property

        Public Property Columns As List(Of ColumnDescription)
            Get
                Return _Columns
            End Get

            Private Set(ByVal value As List(Of ColumnDescription))
                _Columns = value
            End Set
        End Property

        Sub New()
            Dim columns = New List(Of ColumnDescription)()
            For Each employee In NWindContext.Create().Employees
                Dim name As String = employee.FirstName & " " & employee.LastName
                columns.Add(New ColumnDescription(name))
            Next

            SalesByYearData.Columns = columns
        End Sub
    End Module
End Namespace

File: DevExpress-Demo/CS/GridDemo.Wpf/Controls/SalesByYearData.cs
using DevExpress.DemoData.Models;
using System;
using System.Collections.Generic;
using System.Dynamic;

namespace GridDemo {
    public static class SalesByYearData {
        public static List<ExpandoObject> Data {
            get {
                var data = new List<ExpandoObject>();
                Random random = new Random();
                for(int yearIndex = 10; yearIndex > 0; yearIndex--) {
                    int year = DateTime.Now.Year - yearIndex;
                    for(int month = 1; month <= 12; month++) {
                        IDictionary<string, object> row = new ExpandoObject();
                        row["Date"] = new DateTime(year, month, 1);
                        for(int columnIndex = 0; columnIndex < Columns.Count; columnIndex++)
                            row[Columns[columnIndex].PropertyName] = random.Next(30000);
                        data.Add((ExpandoObject)row);
                    }
                }
                return data;
            }
        }
        public static List<ColumnDescription> Columns { get; private set; }
        static SalesByYearData() {
            var columns = new List<ColumnDescription>();
            foreach(var employee in NWindContext.Create().Employees) {
                string name = employee.FirstName + " " + employee.LastName;
                columns.Add(new ColumnDescription(name));
            }
            Columns = columns;
        }
    }
}
